example/service/s3/getObjectWithProgress/getObjectWithProgress.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+// progressWriter creates a progressbar using the cheggaaa/pb library
+// whilst also following the progress using the bytes written to the
+// writerAt object to track its progress
+type progressWriter struct {
+	writer  io.WriterAt
+	progressBar *pb.ProgressBar
```

Comment:
I think we'd prefer not adding this dependency to the example. Having the dependency in the example will cause users `go get` the SDK to also retrieve this dependency, regardless if they have the `example` build tag specified.

I think this example should implement a custom WriterAt, like being done on line 30, but instead of using a progress bar just print out the percentage bytes downloaded. Similar to the [PutObject example](https://github.com/aws/aws-sdk-go/blob/master/example/service/s3/putObjectWithProcess/putObjWithProcess.go#L33-L38), except without the divide. (there is a bug in that example, #2468)

The example can have a comment about how a progress bar could also be used, but I don't think we want to add the direct dependency on the progress bar's package to the SDK.

Reply:
Makes sense, I tried to copy the code and it's showing progress but I think I'm missing some math somewhere:

```go
type progressWriter struct {
	writer  io.WriterAt
	size    int64
	written int64
}

func (pw *progressWriter) WriteAt(p []byte, off int64) (int, error) {

	atomic.AddInt64(&pw.written, int64(off))

	log.Printf("File size:%d progress:%d \n", pw.size, pw.written)

	return pw.writer.WriteAt(p, off)
}

// created like this
writer := &progressWriter{writer: temp, size: size, written: 0}
```

This is showing progress but the numbers are completely off:

```
2019/02/21 11:25:43 File size:35943530 progress:74203673994
2019/02/21 11:25:43 File size:35943530 progress:74219346910
2019/02/21 11:25:43 File size:35943530 progress:74235036186
2019/02/21 11:25:43 File size:35943530 progress:74250726510
2019/02/21 11:25:43 File size:35943530 progress:74266433194
2019/02/21 11:25:43 File size:35943530 progress:74282140926
2019/02/21 11:25:43 File size:35943530 progress:74297865018
2019/02/21 11:25:43 File size:35943530 progress:74313590158
```

Any ideas on how I can fix this?